packages/unified-pipeline/src/utils/ParserUtils.tsx

```diff
@@ -142,21 +143,28 @@ const getChildNodes = (stage: Record<string, any>): Node[] => {
   return childNodes
 }
 
+function getIconAndDisplayName(step: Record<string, any>, stepIndex: number, isV0yaml?: boolean) {
+  //🚨 check the purpose of isV0yaml before merging and add the necessary changes
```

Comment:
This is yet to be discussed with @vardanbansal-harness before merging this PR

Reply:
I think @shaurya-harness might have added this to reduce overall code duplication. Some of the V0 and V1 pipeline yaml parser logic is common. This boolean might be to specify the yaml version.